pinot-broker/src/main/java/org/apache/pinot/broker/queryquota/HelixExternalViewBasedQueryQuotaManager.java

```diff
@@ -323,11 +321,43 @@ public void processQueryQuotaChange() {
       }
       int onlineBrokerCount = otherOnlineBrokerCount + 1;
 
-      double overallRate = Double.parseDouble(quotaConfig.getMaxQueriesPerSecond());
+      // Get stat from property store
+      String tableConfigPath = constructTableConfigPath(tableNameWithType);
+      Stat stat = _propertyStore.getStat(tableConfigPath, AccessOption.PERSISTENT);
+      if (stat == null) {
+        LOGGER.info("Table {} gets deleted from property store. Removing its rate limit.", tableNameWithType);
+        it.remove();
+        continue;
+      }
+
+      // If number of online brokers and table config don't change, there is no need to re-calculate the dynamic rate.
+      if (onlineBrokerCount == queryQuotaEntity.getNumOnlineBrokers() && stat.getVersion() == queryQuotaEntity
+          .getTableConfigStatVersion()) {
+        continue;
```

Comment:
It may be useful to add a log line here with the version

Reply:
It'll flush the log as there might be a lot of tables having qps quotas and their dynamic rate limiters don't need to be changed.